Create SPI and Provider for Verifiable Credentials Signing #25937

wistefan opened the pull request:
The PR adds implementations for SPI and Provider factories and interfaces to prepare the implementation of the https://github.com/keycloak/keycloak/issues/25936 feature.

Closes #25937

MikeTangoEcho committed:
Add X509 thumbprint to JWT when using private_key_jwt

Closes keycloak#12946

Signed-off-by: MikeTangoEcho <mathieu.thine@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

lexcao committed:
Use email verification instead of executing action for `send-verify-email` endpoint

Closes #15190

Add support for `send-verify-email` endpoint to use the `email-verification.ftl` instead of `executeActions.ftl`

Also introduce a new parameter `lifespan` to be able to override the default lifespan value (12 hours)

Signed-off-by: Lex Cao <lexcao@foxmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Commit:
Enhance error logs and error events during UserInfo endpoint and Token Introspection failure

Closes #24344

Signed-off-by: cgeorgilakis-grnet <cgeorgilakis@admin.grnet.gr>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Commit:
Check if UMA is enabled on resource, if not reject the request.

Closes #24422

Signed-off-by: Felix Gustavsson <felix.gustavsson@topgolf.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

douglaspalmer committed:
Deleting a User or User Group might cause that all users suddenly get the permissions of the deleted user.

Closes #24651
Signed-off-by: Douglas Palmer <dpalmer@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

skabano committed:
﻿Enhancing Lightweight access token M2(keycloak#25716)

Closes keycloak#23724

Signed-off-by: shigeyuki kabano <shigeyuki.kabano.sj@hitachi.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

remituveri committed:
Add a toggle to use context attributes on the regex policy provider

Signed-off-by: remi <remi.tuveri@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mposolda committed:
Enable user profile by default
closes #25151

Signed-off-by: mposolda <mposolda@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

rmartinc committed:
Sanitize logs in JBossLoggingEventListenerProvider
Closes #25078

Signed-off-by: rmartinc <rmartinc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

skabano committed:
Creating documentation for Lightweight access token(#25743)

Closes keycloak#23725

Signed-off-by: shigeyuki kabano <shigeyuki.kabano.sj@hitachi.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

rmartinc committed:
Avoid returning duplicated users in LDAP and unsynced
Closes #24141

Signed-off-by: rmartinc <rmartinc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

atharvakshi committed:
Fix for empty realm name issue

Throw ModelException if name is empty when creating/updating a realm

Closes #17449

Signed-off-by: atharva kshirsagar <atharva4894@gmail.com>
Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Co-authored-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Commit:
OIDC Protocol Mappers with same claim

Closes #25774

Signed-off-by: cgeorgilakis-grnet <cgeorgilakis@admin.grnet.gr>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Stabilizing test cases by adding cleanups

Closes #24651

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

pedroigor committed:
Using a valid URI when deleting cookies before/after running tests
Closes #22691

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

alice-wondered committed:
Add logging to the policy providers for general debugging purposes

Signed-off-by: Alice W <105500542+alice-wondered@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Making metrics with labels for embedded Infinispan the default

Closes #25935

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

wistefan committed:
add spi, provider and interfaces for oid4vc #25937

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

reda-alaoui committed:
On invalid submission, IdpUsernamePasswordForm sends back the user to the standard UsernamePasswordForm template

Signed-off-by: Réda Housni Alaoui <reda-alaoui@hey.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

wistefan committed:
update to java 17

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Make passing parameters to `generateEncodedPath()` type-safe (#25944)

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

hmlnarik committed:
Do not switch to admin console too early (#25952)

Closes: #25951

Signed-off-by: Hynek Mlnarik <hmlnarik@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump rollup

Bumps [rollup](https://github.com/rollup/rollup) from 4.9.2 to 4.9.4.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.9.2...v4.9.4)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/react from 18.2.46 to 18.2.47 in /js

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.2.46 to 18.2.47.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 6.17.0 to 6.18.0 in /js

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.17.0 to 6.18.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.18.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump vite from 5.0.10 to 5.0.11 in /js

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.10 to 5.0.11.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.11/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump rollup from 4.9.2 to 4.9.4 in /js

Bumps [rollup](https://github.com/rollup/rollup) from 4.9.2 to 4.9.4.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.9.2...v4.9.4)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin from 6.17.0 to 6.18.0 in /js

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.17.0 to 6.18.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.18.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump jsdom from 23.0.1 to 23.2.0 in /js (#25970)

Bumps [jsdom](https://github.com/jsdom/jsdom) from 23.0.1 to 23.2.0.
- [Release notes](https://github.com/jsdom/jsdom/releases)
- [Changelog](https://github.com/jsdom/jsdom/blob/main/Changelog.md)
- [Commits](https://github.com/jsdom/jsdom/compare/23.0.1...23.2.0)

---
updated-dependencies:
- dependency-name: jsdom
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/node from 20.10.6 to 20.10.7 in /js

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.10.6 to 20.10.7.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Commit:
Ensure protocol forced reauthentication is correctly mapped during SAML identity brokering

Closes #25980

Signed-off-by: Patrick Hamann <patrick@fastly.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

tondrusk committed:
Update web element of the LinkedIn login page (#25905)

Signed-off-by: Tomas Ondrusko <tondrusk@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

potterhe committed:
account console locale zh-CN (#25828)

Signed-off-by: potterhe <potter.he@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump i18next from 23.7.15 to 23.7.16 in /js (#25923)

Bumps [i18next](https://github.com/i18next/i18next) from 23.7.15 to 23.7.16.
- [Release notes](https://github.com/i18next/i18next/releases)
- [Changelog](https://github.com/i18next/i18next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/i18next/i18next/compare/v23.7.15...v23.7.16)

---
updated-dependencies:
- dependency-name: i18next
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Add release announcement and migration for new welcome theme (#25895)

Closes #25894

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

hmlnarik committed:
Fix mvn clean of js/ directory

Fixes: #26021

Signed-off-by: Hynek Mlnarik <hmlnarik@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

agagancarczyk committed:
Localization tabs (#25532)

* Add new localization tabs to Administration Console

Closes #23057

Signed-off-by: Agnieszka <agancarc@redhat.com>
Signed-off-by: Jon Koops <jonkoops@gmail.com>

* css cleanup

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* css cleanup

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

---------

Signed-off-by: Agnieszka <agancarc@redhat.com>
Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Co-authored-by: Jon Koops <jonkoops@gmail.com>
Co-authored-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

hmlnarik committed:
Use proper attribute name in UI

Fixes: #25827

Signed-off-by: Hynek Mlnarik <hmlnarik@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ReneZeidler committed:
Add German translations for account console v3 (#25957)

Signed-off-by: René Zeidler <rene.zeidler@gmx.de>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

shawkins committed:
fix: adds a general error message and descriptions for some exceptions (#25806)

closes: #25746

Signed-off-by: Steve Hawkins <shawkins@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
removed use of ui-ext in library (#25611)

fixes: #25417

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

sschu committed:
Fixed tiny doc typo (#26012)

Signed-off-by: Sebastian Schuster <sebastian.schuster@bosch.io>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

rmartinc committed:
Escape action in the form_post.jwt and only decode path in RedirectUtils (#93) (#25995)

Closes #90

Signed-off-by: rmartinc <rmartinc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

McLaynV committed:
Update cs+sk login messages - fixes #25962 (#25963)

Signed-off-by: Vojtěch Miklín <mclayn.v@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Add the build step to the overall status check (#25982)

Closes #25981

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

pedroigor committed:
Improve brute force documentation around how the effective wait time is calculated

Closes #25915

Signed-off-by: Pedro Igor <pigor.craveiro@gmail.com>
Co-authored-by: Alexander Schwartz <alexander.schwartz@gmx.net>
Co-authored-by: andymunro <48995441+andymunro@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
fixed label on resource (#25623)

fixes: #20304

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
remove userProfileMetadata from user (#25608)

fixes: #25463

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin from 6.18.0 to 6.18.1 in /js (#26010)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.18.0 to 6.18.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.18.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump react-hook-form from 7.49.2 to 7.49.3 in /js (#26009)

Bumps [react-hook-form](https://github.com/react-hook-form/react-hook-form) from 7.49.2 to 7.49.3.
- [Release notes](https://github.com/react-hook-form/react-hook-form/releases)
- [Changelog](https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md)
- [Commits](https://github.com/react-hook-form/react-hook-form/compare/v7.49.2...v7.49.3)

---
updated-dependencies:
- dependency-name: react-hook-form
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump aquasecurity/trivy-action from 0.16.0 to 0.16.1 (#25885)

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.16.0 to 0.16.1.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/91713af97dc80187565512baba96e4364e983601...d43c1f16c00cfd3978dde6c07f4bbcf9eb6993ca)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mabartos committed:
Could not format default value for log formats (#25910)

Closes #25908

Signed-off-by: Martin Bartoš <mabartos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
change text based on the context (#25664)

fixes: #25302

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
fixed message for disabling clients (#25672)

fixes: #25636

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Fix OfflineServletAdapterTest failures, and improve logging (#25724)

Closes #25714
Closes #14448

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

andymunro committed:
Clarify note about containers

Closes #26006

Signed-off-by: AndyMunro <amunro@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Add landmark element for loading state for consoles (#26037)

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Remove concurrent loading of remote sessions as at startup time only one node is up anyway. (#25709)

Closes #22082

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Co-authored-by: Martin Kanis <martin-kanis@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

agagancarczyk committed:
Added welcome tab to admin-ui (#25992)

* resolved conflict

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed space

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* reemoved logo from welcome tab

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* feedback

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed mashead tests

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed mashead tests

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

---------

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Co-authored-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

shawkins committed:
doc: adding notes about header priority (#25959)

closes: keycloak#23023

Signed-off-by: Steve Hawkins <shawkins@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
use content json also for own menu (#25744)

* use content json also for own menu

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

* change isHidden with isVisible

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

---------

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/node from 20.10.7 to 20.10.8 in /js (#26062)

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.10.7 to 20.10.8.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @rollup/plugin-typescript from 11.1.5 to 11.1.6 in /js (#26061)

Bumps [@rollup/plugin-typescript](https://github.com/rollup/plugins/tree/HEAD/packages/typescript) from 11.1.5 to 11.1.6.
- [Changelog](https://github.com/rollup/plugins/blob/master/packages/typescript/CHANGELOG.md)
- [Commits](https://github.com/rollup/plugins/commits/typescript-v11.1.6/packages/typescript)

---
updated-dependencies:
- dependency-name: "@rollup/plugin-typescript"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

hmlnarik committed:
Add progressbar role (#25988)

Fixes: #25951

Signed-off-by: Hynek Mlnarik <hmlnarik@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
more missing translations (#26018)

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
remove default so it can be empty (#26024)

fixes: #23683

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 6.18.0 to 6.18.1 in /js (#26008)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.18.0 to 6.18.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.18.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Reduce internal unsupported options in the Keycloak HA documentation

Closes #26068

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

reda-alaoui committed:
Add federated identity ProviderEvent(s)

Signed-off-by: Réda Housni Alaoui <reda-alaoui@hey.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

rmartinc committed:
Integrate registration with terms and conditions required action
Closes #25891

Signed-off-by: rmartinc <rmartinc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Remove conditionals about Linux vs. Windows (#26031)

Closes #26028

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Aboullos committed:
Add tests to Cypress (#25958)

Signed-off-by: Alfredo Moises Boullosa <aboullos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

andymunro committed:
Change RHDG to Infinispan

Closes #26083

Signed-off-by: AndyMunro <amunro@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Replace `react-error-boundary` with own implementation (#26094)

Closes #21515

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump eslint-plugin-prettier from 5.1.2 to 5.1.3 in /js

Bumps [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) from 5.1.2 to 5.1.3.
- [Release notes](https://github.com/prettier/eslint-plugin-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-plugin-prettier/compare/v5.1.2...v5.1.3)

---
updated-dependencies:
- dependency-name: eslint-plugin-prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

cdevienne committed:
Fix rest provider example for kc >= 23

Starting with KC23, a @Path is required for an exposed method to work properly

Fixes #26088

Signed-off-by: Christophe de Vienne <christophe.devienne@orus.io>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

KevinMartinsDevUbi committed:
Complete the documentation for the use case of a resource from an email template. (#25705)

Signed-off-by: Kevin MARTINS <k.martins@ubitransport.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump wireit from 0.14.1 to 0.14.3 in /js

Bumps [wireit](https://github.com/google/wireit) from 0.14.1 to 0.14.3.
- [Changelog](https://github.com/google/wireit/blob/main/CHANGELOG.md)
- [Commits](https://github.com/google/wireit/compare/v0.14.1...v0.14.3)

---
updated-dependencies:
- dependency-name: wireit
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

thomasdarimont committed:
Avoid calling primitive wrapper constructors in server-spi-private (#24163)

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

lrkwz committed:
Update placeholders for hostname and port (#24153)

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Co-authored-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ReneZeidler committed:
Remove unnecessary tooltip on device activity page (#26078)

Closes #26032

Signed-off-by: René Zeidler <rene.zeidler@gmx.de>
Co-authored-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump vitest from 1.1.1 to 1.1.3 in /js (#25969)

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 1.1.1 to 1.1.3.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.1.3/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Have a more descriptive error message when infinite recusion happens (#26043)

Closes #21151

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

agagancarczyk committed:
Fixed a11y violations (#26125)

* fixed a11y violations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed a11y violations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed a11y violations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed a11y violations

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

---------

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Co-authored-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Promote Account Console v3 to default and deprecate v2 (#25852)

Closes #19663

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Co-authored-by: Martin Bartoš <mabartos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump react-router-dom from 6.21.1 to 6.21.2 in /js

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 6.21.1 to 6.21.2.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@6.21.2/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/node from 20.10.8 to 20.11.0 in /js

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.10.8 to 20.11.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Remove `lodash` as a hard dependency from the Admin Client (#26136)

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
refactored authentication to use the ui-shared (#25730)

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

agagancarczyk committed:
fixed critical a11y violation (#26157)

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Co-authored-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

shawkins committed:
fix: switching the raw distribution to a weak readiness check (#26097)

also adding a thread dump if the server doesn't seem to stop properly

closes: #23786

Signed-off-by: Steve Hawkins <shawkins@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mabartos committed:
Do not use raw parameterized PropertyMapper (#26093)

Closes #26092

Signed-off-by: Martin Bartoš <mabartos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

jonkoops committed:
Add `@Deprecated` annotation to deprecated profiles (#26174)

Signed-off-by: Jon Koops <jonkoops@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump rollup

Bumps [rollup](https://github.com/rollup/rollup) from 4.9.4 to 4.9.5.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.9.4...v4.9.5)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump vitest from 1.1.3 to 1.2.0 in /js

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 1.1.3 to 1.2.0.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.2.0/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump prettier from 3.1.1 to 3.2.2 in /js

Bumps [prettier](https://github.com/prettier/prettier) from 3.1.1 to 3.2.2.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/3.1.1...3.2.2)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Changing to the Infinispan BOM to avoid mis-aligned Infinispan dependencies (#26137)

Closes #22922

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Pedro Ruivo <pruivo@redhat.com>
Co-authored-by: Pedro Ruivo <pruivo@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump rollup from 4.9.4 to 4.9.5 in /js

Bumps [rollup](https://github.com/rollup/rollup) from 4.9.4 to 4.9.5.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v4.9.4...v4.9.5)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
added attributes to KeycloakProfile definition (#25617)

* added attributes to KeycloakProfile definition

fixes: #25403
Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

* Update js/libs/keycloak-js/dist/keycloak.d.ts

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>

---------

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mabartos committed:
Comprehensive log for the LoggingDistTest and Quarkus IT (#26199)

Closes #26198

Signed-off-by: Martin Bartoš <mabartos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

elkman committed:
fix non-POSIX usage of expr in kc.sh (#26166)

replaces \+ by \{1,\} in regular expression for better POSIX compatibility

Signed-off-by: elkman <elkman@users.noreply.github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

shawkins committed:
fix: provide better error messages for list options (#25918)

closes: #25235

Signed-off-by: Steve Hawkins <shawkins@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/react from 18.2.47 to 18.2.48 in /js

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.2.47 to 18.2.48.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin from 6.18.1 to 6.19.0 in /js

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.18.1 to 6.19.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.19.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump vite-plugin-dts from 3.7.0 to 3.7.1 in /js

Bumps [vite-plugin-dts](https://github.com/qmhc/vite-plugin-dts) from 3.7.0 to 3.7.1.
- [Release notes](https://github.com/qmhc/vite-plugin-dts/releases)
- [Changelog](https://github.com/qmhc/vite-plugin-dts/blob/main/CHANGELOG.md)
- [Commits](https://github.com/qmhc/vite-plugin-dts/compare/v3.7.0...v3.7.1)

---
updated-dependencies:
- dependency-name: vite-plugin-dts
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 6.18.1 to 6.19.0 in /js

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.18.1 to 6.19.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.19.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump @types/node from 20.11.0 to 20.11.3 in /js

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.11.0 to 20.11.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

ahus1 committed:
Deprecating the offline session preloading (#26160)

Closes #25300

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

dependabot[bot] committed:
Bump lightningcss from 1.22.1 to 1.23.0 in /js

Bumps [lightningcss](https://github.com/parcel-bundler/lightningcss) from 1.22.1 to 1.23.0.
- [Release notes](https://github.com/parcel-bundler/lightningcss/releases)
- [Commits](https://github.com/parcel-bundler/lightningcss/compare/v1.22.1...v1.23.0)

---
updated-dependencies:
- dependency-name: lightningcss
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
fixes navigation when searching for groups (#26053)

fixes: #25927

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

shawkins committed:
addendum to #23786 - readiness check should end after the first dump (#26215)

Signed-off-by: Steve Hawkins <shawkins@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

agagancarczyk committed:
Localization: Realm Overrides Fixes (#26169)

* resolved conflict

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* improvements

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* improved tests

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* feedback

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* test fix

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* test fix

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* resolved conflict

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

* fixed test

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>

---------

Signed-off-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Co-authored-by: Agnieszka Gancarczyk <agancarc@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

edewit committed:
added choose as an option to reset (#26065)

fixes: #25939

Signed-off-by: Erik Jan de Wit <erikjan.dewit@gmail.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

pkeuter committed:
fix: permissions tab should be only shown with the correct rights (#26034)

Signed-off-by: Peter Keuter <github@peterkeuter.nl>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

BlackVoid committed:
Add login/ui (#26218)

Signed-off-by: Felix Gustavsson <felix.gustavsson@topgolf.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

Aboullos committed:
Fix flaky client scopes test for Firefox (#26195)

Closes #26148

Signed-off-by: Alfredo Moises Boullosa <aboullos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mabartos committed:
Do not use general debug log level for tests (#26242)

Closes #26241

Signed-off-by: Martin Bartoš <mabartos@redhat.com>
Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

mposolda commented on services/src/main/java/org/keycloak/protocol/oid4vc/issuance/VCIssuerException.java:
```diff
@@ -0,0 +1,17 @@
+ */
+public class VCIssuerException extends RuntimeException {
+
+	public VCIssuerException(String message) {
```
Minor: Is it possible to use same padding like other Keycloak classes? So 4 characters only in java classes?

wistefan replied:
fixed by #f74153f

mposolda commented on services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/VCSigningServiceSpi.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public Class<? extends ProviderFactory> getProviderFactoryClass() {
+        return VCSigningServiceProviderFactory.class;
```
Is it possible to make sure that the class referenced from here is an interface rather than abstract class? It currently works with an abstract class, but AFAIK it is not recommended pattern and it is possible this can cause issues in the future (if Keycloak later does some refactoring in providers initialization part etc).

Hopefully just something minor like having interface `VCSigningServiceProviderFactory` and rename current abstract class to `AbstractVCSigningServiceProviderFactory` will do the job?

wistefan replied:
fixed by [#7d5dea4](https://github.com/keycloak/keycloak/pull/25947/commits/7d5dea418e6cd92a3cb7b7577a545116a9e49ce8)

mposolda reviewed (commented):
Thanks! Added 2 minor comments inline.

Additional comment: Is it possible to introduce `EXPERIMENTAL` feature `OID4VC` into the `org.keycloak.common.Profile` ? Then also make sure that your factories (jwtSigningServiceProviderFactory, LDSigningServiceProviderFactory, VCSigningServiceProviderFactory) all implement `EnvironmentDependentFactory` and are supported just when the feature is enabled?

I appreciate the factories are not yet enabled anyway, but I would appreciate them to be "hidden" behind the feature flag from day 1, so we don't forget at it later.

NOTE: This will require some minor changes in few tests like `ProfileTest`  and some `help*` files used during quarkus distribution tests. For example see this commit for the inspiration https://github.com/keycloak/keycloak/commit/7da52a43 ,

wistefan commented:
> Thanks! Added 2 minor comments inline.
> 
> Additional comment: Is it possible to introduce `EXPERIMENTAL` feature `OID4VC` into the `org.keycloak.common.Profile` ? Then also make sure that your factories (jwtSigningServiceProviderFactory, LDSigningServiceProviderFactory, VCSigningServiceProviderFactory) all implement `EnvironmentDependentFactory` and are supported just when the feature is enabled?
> 
> I appreciate the factories are not yet enabled anyway, but I would appreciate them to be "hidden" behind the feature flag from day 1, so we don't forget at it later.

Hi, yes I will add it. Was just not aware of it.

mposolda commented:
@tnorimat @thomasdarimont Anyone of you can also review this please?

wistefan committed:
fix indention

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

wistefan committed:
make provider factory an interface

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

wistefan committed:
add feature flag

Signed-off-by: Stefan Wiedemann <wistefan@googlemail.com>

wistefan commented:
> > Thanks! Added 2 minor comments inline.
> > Additional comment: Is it possible to introduce `EXPERIMENTAL` feature `OID4VC` into the `org.keycloak.common.Profile` ? Then also make sure that your factories (jwtSigningServiceProviderFactory, LDSigningServiceProviderFactory, VCSigningServiceProviderFactory) all implement `EnvironmentDependentFactory` and are supported just when the feature is enabled?
> > I appreciate the factories are not yet enabled anyway, but I would appreciate them to be "hidden" behind the feature flag from day 1, so we don't forget at it later.
> 
> Hi, yes I will add it. Was just not aware of it.

Fixed by [#1e5d94d](https://github.com/keycloak/keycloak/pull/25947/commits/1e5d94d2acfbcc16119b1e952644016249c6ab0c)

Commit:
Merge branch 'main' into 25937-create-spi-and-provider-for-verifiable-credentials-signing

jonkoops commented:
Closing this PR as a bunch of unrelated commits have been added. Feel free to open up a new one so the reviewers are reset.
